src/nl/hannahsten/texifyidea/refactoring/inlinecommand/LatexInlineCommandHandler.kt

```diff
@@ -0,0 +1,131 @@
+        return language == LatexLanguage
+    }
+
+    override fun canInlineElement(element: PsiElement?): Boolean {
+        return canInlineLatexElement(element)
+    }
+
+    override fun inlineElement(project: Project, editor: Editor?, element: PsiElement) {
+        // Resolve the file to be inlined
+        val inlineCommand: LatexCommands = resolveInlineCommandDefinition(element) ?: return
+        val myReference = getReference(element, editor)
+
+        val dialog = LatexInlineCommandDialog(
+            project,
+            inlineCommand,
+            myReference,
+            editor != null && inlineCommand.firstChildOfType(LatexCommands::class) != myReference
+        )
+
+        if (dialog.myOccurrencesNumber > 0) {
+            if (ApplicationManager.getApplication().isUnitTestMode) {
+                try {
+                    dialog.doAction()
+                }
+                finally {
+                    dialog.close(DialogWrapper.OK_EXIT_CODE, true)
+                }
+            }
+            else {
+                dialog.show()
+            }
+        }
+        else {
+            Notification(
+                "LaTeX",
+                "No usages found",
+                "Could not find any usages for ${inlineCommand.name}",
+                NotificationType.ERROR
+            ).notify(project)
+        }
+    }
+
+    companion object {
+
+        /**
+         * This is static only so that the unit tests can use it, and also since it can be static
+         */
+        fun canInlineLatexElement(element: PsiElement?): Boolean {
+            if (element !is LatexCommands) {
+                return false
+            }
+
+            val file = element.containingFile
+            val files: MutableSet<PsiFile> = HashSet(file.referencedFileSet())
+
+            val cmds = getCommandsInFiles(files, file)
+
+            val candiateUserCommands = cmds.filter { it.isDefinition() && !it.isEnvironmentDefinition() }
+
+            val canInline = candiateUserCommands.any {
+                LatexCommandsAndEnvironmentsCompletionProvider.getDefinitionName(it) == element.name
+            }
+            return canInline
+        }
+
+        fun resolveInlineCommandDefinition(element: PsiElement): LatexCommands? {
+            val file = element.containingFile
+            val files: MutableSet<PsiFile> = HashSet(file.referencedFileSet())
+
+            val cmds = getCommandsInFiles(files, file)
+
+            val candiateUserCommands = cmds.filter { it.isDefinition() && !it.isEnvironmentDefinition() }
+
+            val thisCommandDefinitions = candiateUserCommands.filter {
+                LatexCommandsAndEnvironmentsCompletionProvider.getDefinitionName(it) == (element as LatexCommands).name
+            }
+            if (thisCommandDefinitions.size != 1)
+                Log.warn("Did not find a singleton definition, found " + thisCommandDefinitions.size)
```

Comment:
The warning is there to say "hey, I'm guessing at which one to use" so if there is a utility to resolve the actual definition instead of borrowed digging, that would be cool

Reply:
Sure but the user won't see the warning until he opens idea.log, at which point it is probably unclear at all what part of TeXiFy the warning is about unless you're actually debugging it, in which case an info level logging would be good enough.
If you want to alert the user, use an info-level Notification, but in this case I think it's okay to keep silent

Reply:
Is there a util command to just resolve the current active definition?

Reply:
Not sure what you mean with "current active", the definition reference will probably just point to the last (re)definition, which is good enough for me

Reply:
That isnt guaranteed, but if you're fine with it, then I wont lose sleep over it

Reply:
Nevermind, losing sleep; 

```latex
\newcommand{\sail}[2][ha]{#2 on boat}
\newcommand{\ahoy}[1]{#1 pirate, \sail{#1}}

\begin{document}
    \ahoy{me}
    \ahoy{Arggg}

    \input{argg}
\end{document}
```

argg.tex:
```latex
\renewcommand{\ahoy}[1]{#1 matey, \sail{#1}}

\ahoy{me}
\ahoy{Arggg}
```

You should be able to notice that it is just terrible in this case

Reply:
Of course, but TeXiFy has no notion of ordering of command definitions in a document, so I would say that falls outside the scope of this PR.

Reply:
If you say so...

Reply:
I do. In practice that means I tend to wait until some user complains about it before I fix it :smile: 
The advantage of that is that I automatically spend more time in more often used features.